addons/binding/org.openhab.binding.elkm1/src/main/java/org/openhab/binding/elkm1/internal/elk/message/ZoneChangeUpdate.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.elkm1.internal.elk.message;
+
+import org.openhab.binding.elkm1.internal.elk.ElkCommand;
+import org.openhab.binding.elkm1.internal.elk.ElkMessage;
+import org.openhab.binding.elkm1.internal.elk.ElkZoneConfig;
+import org.openhab.binding.elkm1.internal.elk.ElkZoneStatus;
+
+/**
+ * Called by the elk when the zone changes.
+ *
+ * @author David Bennett - Initial Contribution
+ *
+ */
+public class ZoneChangeUpdate extends ElkMessage {
+    ElkZoneConfig config;
+    ElkZoneStatus status;
+    int zoneNumber;
+
+    public ZoneChangeUpdate(String incomingData) {
+        super(ElkCommand.ZoneChangeUpdateReport);
+
+        String zoneNumberStr = incomingData.substring(0, 3);
+        zoneNumber = Integer.valueOf(zoneNumberStr, 16);
```

Comment:
This needs to be straight decimal, not hex per the doc (and per my testing, my zone 17 came up as 23):
`zoneNumber = Integer.valueOf(zoneNumberStr);`

https://www.fixagate.com/assets/m1_rs232_ascii_protocol.pdf
section 4.39

This is the only critical issue I found in the impl outside of the discovery zero check you already fixed.